### ISSUE TITLE
Regression: Unstable formatting on long strings

### ISSUE THREAD

**Issue by JelleZijlstra** [psf/black#1042]
I tested Black master on our codebase in preparation for a release, but found that Black failed to format 17 files (the previous release didn't crash on any of our files). It's not clear that all of these have the same root cause, but here's one I minimized somewhat:

```
$ black infra-common/qroxy/qroxy.py 
error: cannot format infra-common/qroxy/qroxy.py: INTERNAL ERROR: Black produced different code on the second pass of the formatter.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: /tmp/blk_i040l_hj.log
Oh no! 💥 💔 💥
1 file failed to reformat.
$ cat /tmp/blk_i040l_hj.log 
--- source
+++ first pass
@@ -1,10 +1,10 @@
 async def show_status():
     while True:
         try:
             if report_host:
                 data = (
-                    f"{boxname},{alive_server_connections},{server_queue.size()},{requester_queue.size()}"
-                ).encode()
+                    f"{boxname},{alive_server_connections},{server_queue.size()},{requester_queue.size()}".encode()
+                )
         except Exception as e:
             log.error(f"show_status thread encounters exception: {e}")
 
--- first pass
+++ second pass
@@ -1,10 +1,8 @@
 async def show_status():
     while True:
         try:
             if report_host:
-                data = (
-                    f"{boxname},{alive_server_connections},{server_queue.size()},{requester_queue.size()}".encode()
-                )
+                data = f"{boxname},{alive_server_connections},{server_queue.size()},{requester_queue.size()}".encode()
         except Exception as e:
             log.error(f"show_status thread encounters exception: {e}")
 
$ cat infra-common/qroxy/qroxy.py 
async def show_status():
    while True:
        try:
            if report_host:
                data = (
                    f"{boxname},{alive_server_connections},{server_queue.size()},{requester_queue.size()}"
                ).encode()
        except Exception as e:
            log.error(f"show_status thread encounters exception: {e}")

```

**Comment by hugovk** [comment]
Like https://github.com/psf/black/issues/1044, git bisect points to the same 957ba24bb6cdd32e4dd14ff2808dcf4d58851844 in PR #850 to fix #548:

```
957ba24bb6cdd32e4dd14ff2808dcf4d58851844 is the first bad commit
commit 957ba24bb6cdd32e4dd14ff2808dcf4d58851844
Date:   Wed May 15 21:11:04 2019 -0700

    remove obviously unnecessary parentheses (#850)

    Fixes #548

:100644 100644 17aea7af264adafc4f5ee18cb9389cdfcdce8e17 694c1e961aa41bef489ea08145f1ee9faed7c69b M	black.py
:040000 040000 8da77aae59466f50234b0c16138cfcc5e88ebc17 3d7f6604495d23cbe68f30c434cf0d7a8ad1779d M	tests
bisect run success
```

Again, the second pass is stable and leads to no further formatting.

**Comment by JelleZijlstra** [comment]
Sounds like we should back out that change.
